torch/_dynamo/skipfiles.py

```diff
@@ -62,119 +145,90 @@ def _module_dir(m: types.ModuleType):
+    _module_dir(torch) + "distributed/_functional_collectives.py",
+    torch.distributions.normal.__file__,
+    torch.distributions.independent.__file__,
+    torch.distributions.utils.__file__,
+    torch.utils._contextlib.__file__,
+    torch.fx._pytree.__file__,
 }
 
 if torch.distributed.is_available():
     # Inline the checkpoint code from distributed
     import torch.distributed.algorithms._checkpoint.checkpoint_wrapper
 
-    FILENAME_ALLOWLIST |= {
+    FILENAME_INLINELIST |= {
         torch.distributed.algorithms._checkpoint.checkpoint_wrapper.__file__
     }
 
 # Include optimizer code for tracing
-FILENAME_ALLOWLIST |= {
+FILENAME_INLINELIST |= {
     inspect.getfile(obj)
     for obj in torch.optim.__dict__.values()
     if inspect.isclass(obj)
 }
-FILENAME_ALLOWLIST |= {torch.optim._functional.__file__}
-FILENAME_ALLOWLIST |= {torch.utils._foreach_utils.__file__}
-
-# Do trace through match and replace patterns used in PT2E QAT
-# Note: These patterns are comprised of torch ops and for internal use only.
-# They are exported to aten graphs before being passed to the FX subgraph rewriter.
-# TODO: find a better way to express this path without having to import
-# `torch.ao.quantization.pt2e`, which interferes with memory profiling
-FILENAME_ALLOWLIST |= {
-    _module_dir(torch) + "ao/quantization/pt2e/qat_utils.py",
-    _module_dir(torch) + "ao/quantization/quantizer/xnnpack_quantizer.py",
-    _module_dir(torch) + "ao/quantization/pt2e/representation/rewrite.py",
-    _module_dir(torch) + "ao/quantization/pt2e/utils.py",
-    _module_dir(torch) + "ao/quantization/pt2e/eval_utils.py",
-}
-
-FILENAME_ALLOWLIST |= {
-    _module_dir(torch) + "_export/constraints.py",
-}
-
-FILENAME_ALLOWLIST |= {
-    _module_dir(torch) + "_higher_order_ops/cond.py",
-}
 
 # TODO (zhxchen17) Make exportdb importable here.
-FILENAME_ALLOWLIST |= set(
+FILENAME_INLINELIST |= set(
     glob.glob(_module_dir(torch) + "_export/db/examples/*.py"),
 ) | {
     _module_dir(torch) + "_export/wrappers.py",
 }
 
-# torch.func: need to allow this file to be able to look at functorch transforms
-FILENAME_ALLOWLIST |= {
-    _module_dir(torch) + "_functorch/apis.py",
-    _module_dir(torch) + "_functorch/deprecated.py",
-}
 
-FILENAME_ALLOWLIST |= {
-    _module_dir(torch) + "distributed/tensor/parallel/_utils.py",
-    _module_dir(torch) + "distributed/tensor/parallel/style.py",
-    _module_dir(torch) + "distributed/tensor/parallel/_data_parallel_utils.py",
-    _module_dir(torch) + "distributed/_tensor/api.py",
-    _module_dir(torch) + "distributed/_tensor/device_mesh.py",
+# Force inline functions under these modules, even the modules is in *_SKIPLIST.
+SUBMODULE_INLINELIST = {
+    torch.nn,
+    torch.distributions,
+    torch.testing,
+    torch.ao.nn,
+    torch._refs,
+    torch._prims,
+    torch._decomp,
+    torch.utils._contextlib,
+    torch.utils._pytree,
+    torch.fx._pytree,
+    torch.sparse,
```

Comment:
It is nice that you chose not to change observable behavior, that is good.

However, it is still non-obvious to me how to reason out that the new code here does the same thing as the old code (and not something subtly different.) Can you walk us through the changes you made?

Reply:
You are right, they may have subtly different. But I think if we refactor the code, we should correct the mistake we made before, and consolidate these rules into a more organized way, which is the goal of the refactor. I expected there will be slightly difference and trigger some bugs, then we should just fix these bugs using the new structure. I'm happy to walk though these changes and potential changes in my mind.